edit/lint.js

```diff
@@ -0,0 +1,424 @@
+      showSavedMessage();
+    } else {
+      showLinterErrorMessage(linter, t('setLinterError'));
+    }
+  });
+  $('.reset', popup).addEventListener('click', event => {
+    event.preventDefault();
+    const linter = checkLinter(event.target.dataset.linter);
+    let rules;
+    if (linter === 'stylelint') {
+      setStylelintRules();
+      rules = stylelintDefaultConfig.rules;
+    } else {
+      setCSSLintRules();
+      rules = csslintDefaultRuleset;
+    }
+    popup.codebox.setValue(stringifyRules(rules));
+    updateLinter(linter);
+  });
+  $('.cancel', popup).addEventListener('click', event => {
+    event.preventDefault();
+    $('.dismiss').dispatchEvent(new Event('click'));
+  });
+}
+
+function openStylelintSettings() {
+  const linter = prefs.get('editor.linter');
+  BG.chromeSync.getValue(
+    linter === 'stylelint'
+      ? 'editorStylelintRules'
+      : 'editorCSSLintRules'
+  ).then(rules => {
+    if (!rules || rules.length === 0) {
+      rules = linter === 'stylelint'
+        ? setStylelintRules(rules)
+        : setCSSLintRules(rules);
+    }
+    const rulesString = stringifyRules(rules);
+    setupLinterPopup(rulesString);
+  });
+}
+
+function setupLinterPopup(rules) {
+  const linter = prefs.get('editor.linter');
+  const linterTitle = linter === 'stylelint' ? 'Stylelint' : 'CSSLint';
+  function makeButton(className, text) {
+    return $element({tag: 'button', className, type: 'button', textContent: t(text), dataset: {linter}});
+  }
+  function makeLink(url, textContent) {
+    return $element({tag: 'a', target: '_blank', href: url, textContent});
+  }
+  function setJSONMode(cm) {
+    cm.setOption('mode', 'application/json');
+    cm.setOption('lint', 'json');
+  }
+  const popup = showCodeMirrorPopup(t('setLinterRulesTitle', linterTitle), $element({
+    appendChild: [
+      $element({
+        tag: 'p',
+        appendChild: [
+          t('setLinterLink') + ' ',
+          makeLink(
+            linter === 'stylelint'
+              ? 'https://stylelint.io/demo/'
+              : 'https://github.com/CSSLint/csslint/wiki/Rules-by-ID',
+            linterTitle
+          ),
+          linter === 'csslint' ? ' ' + t('showCSSLintSettings') : ''
+        ]
+      }),
+      makeButton('save', 'styleSaveLabel'),
+      makeButton('cancel', 'confirmCancel'),
+      makeButton('reset', 'genericResetLabel'),
+      $element({
+        tag: 'span',
+        className: 'saved-message',
+        textContent: t('genericSavedMessage')
+      })
+    ]
+  }));
+  const contents = $('.contents', popup);
+  const loadJSON = window.jsonlint ? [] : [
+    'vendor/codemirror/mode/javascript/javascript.js',
+    'vendor/codemirror/addon/lint/json-lint.js',
+    'vendor/jsonlint/jsonlint.js'
+  ];
+  contents.insertBefore(popup.codebox.display.wrapper, contents.firstElementChild);
+  popup.codebox.focus();
+  popup.codebox.setValue(rules);
+  onDOMscripted(loadJSON).then(() => setJSONMode(popup.codebox));
```

Comment:
Add `popup.codebox.clearHistory();` after codebox.setValue, otherwise it's possible to undo to an empty contents via Ctrl-Z. Leave reset as is - to make it possible to undo a reset.